daffodil-core/src/main/scala/org/apache/daffodil/dsom/Term.scala

```diff
@@ -791,6 +807,125 @@ trait Term
     res
   }.value
 
+  /*
+   * Returns a list of Elements that could follow this Term, within its
+   * lexically enclosing model group.
+   *
+   * Does not follow backpointers from group defs to group refs.
+   *
+   * It is ok, and has no negative impact on sharing for a Term to reference
+   * its lexically enclosing model group.
+   *
+   * Within the lexically enclosing model group, if it is a sequence, then
+   * this computation involves subsequent siblings, children of those siblings.
+   */
+
+  lazy val possibleNextLexicalSiblingElementsInInfoset: PossibleNextElements = {
```

Comment:
At end of review, move this whole algorithm into the runtime1 package. This is very specific to the streaming unparser of runtime1. 

Given that isRequiredInInfoset is renamed to isRequiredSteramingUnpaserEvent, this should be renamed to possibleNextLexicalStreamingUnparserEvent